code/modules/mob/living/carbon/xenomorph/strains/castes/runner/acid.dm

```diff
@@ -47,15 +48,20 @@
 	var/caboom_struct_acid_type = /obj/effect/xenomorph/acid
 
 /datum/behavior_delegate/runner_acider/proc/modify_acid(amount)
+	if(acid_amount >= acid_gen_cap) //This checks if the current acid is greater or equal to the cap, and stops/starts passive acid generation depending if it is.
+		acid_passive_regen = 0
+	else
+		acid_passive_regen = 1
```

Comment:
Rather than variably alter what the regen amount is (even as is the `1` would be better as `initial(acid_passive_regen)`), have this check instead around `modify_acid(acid_passive_regen)` in `/datum/behavior_delegate/runner_acider/on_life()`

In other words, leave `acid_passive_regen` alone and instead only call `modify_acid(acid_passive_regen)` if they meet the treshold.